routes/utils.js

```diff
@@ -30,35 +30,57 @@ router.get('/search', (req, res, next) => {
     // search for item name with given string
     if (results.length <= 0) {
       // get from database
+      let ids = conHelper.getObjectIdByItemName(req.query.itemName);
+      for(let i = 0; i < ids.length; i++){
+        results[i] = conHelper.getObjectById(ids[i]);
+      }
+      return res.status(200).json(results);
+
     } else {
       // filter by results
       results = results.filter(val => val.itemName.toLowerCase()
         .includes(req.query.itemName.toLowerCase()));
+      return res.status(200).json(results);
     }
   }
 
   if (req.query.zipcode) {
     if (results.length <= 0) {
       // get from database
+      let ids = conHelper.getObjectIdByZipcode(req.query.zipcode);
+      for(let i = 0; i < ids.length; i++){
+        results[i] = conHelper.getObjectById(ids[i]);
+      }
+      return res.status(200).json(results);
     } else {
       // filter by results
       results = results.filter(val => val.zipcode.equals(req.query.zipcode));
+      return res.status(200).json(results);
     }
   }
 
   if (req.query.username) {
     if (results.length <= 0) {
       // get from database
+      results = conHelper.getUserListed(req.query.username);
+      return res.status(200).json(results);
     } else {
       // filter by results
       results = results.filter(val => val.username.toLowerCase()
         .includes(req.query.username.toLowerCase()));
+      return res.status(200).json(results);
```

Comment:
Don't return json here- if someone gives both username and first name, this would stop at username and not filter by first name as well